crates/task-impls/src/consensus.rs

```diff
@@ -1237,6 +1259,38 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                     }
                 }
             }
+            HotShotEvent::ViewSyncFinalizeCertificate2Recv(certificate) => {
+                if !certificate.is_valid_cert(self.quorum_membership.as_ref()) {
+                    warn!(
+                        "View Sync Finalize certificate {:?} was invalid",
+                        certificate.get_data()
+                    );
+                    return;
+                }
+
+                self.view_sync_cert = Some(certificate.clone());
+
+                // cancel poll for votes
+                self.quorum_network
+                    .inject_consensus_info(ConsensusIntentEvent::CancelPollForVotes(
+                        *certificate.view_number,
+                    ))
+                    .await;
+
+                debug!(
+                    "Attempting to publish proposal after forming a View Sync Finalized Cert for view {}",
+                    *certificate.view_number
+                );
+
+                let view = certificate.view_number + 1;
+
+                if !self
+                    .publish_proposal_if_able(view, self.timeout_cert.clone(), &event_stream)
```

Comment:
timeout_cert should be `None` here or else we can attach an old timeout cert, which would create an invalid proposal

Reply:
It's possible that both view sync and timeout certs are valid (but really replicas should not be voting for both certs and shouldn't happen) but we only need one for a valid proposal.  So it's ok never include a timeout_cert from this code path.

It probably works just fine because we clear the `self.timeout_cert` after proposing, but might as well make sure a bug can't slip in.